Merge Branch "feature/django-osf" to "develop"

cslzchen opened the pull request:
OSF Ticket: https://openscience.atlassian.net/browse/OSF-7064.

cslzchen committed:
Add `dataSource` settings for OSF PosgreSQL DB.

cslzchen committed:
Implement JPA prototype.
    - DAO interface: OpenScienceFrameworkDao
    - DAO implementation: OpenScienceFrameworkDaoImpl
    - @Entity: OpenScienceFrameworkUser
    - @Entity: OpenScienceFrameworkTimeBasedOneTimePassword

cslzchen committed:
Update spring config to handle two databases.

cslzchen committed:
Update `authenticationHandler`: [skip ci]
    - refactor into package io.cos.cas.adaptors.postgres.handlers
    - replace `MongoOperations` with `OpenScienceFrameworkDaoImpl`
    - update spring config

cslzchen committed:
Remove unnecessary `this` reference in class.

cslzchen committed:
Add `institution` to JPA prototype:
    -update spring config
    - update OSF DAO
    - @Entity: OpenScienceFrameworkInstitution

cslzchen committed:
Update postgres support for institution logout: [skip ci]
    - logoutHandler -> OpenScienceFrameworkInstitutionHandler
    - logoutManager.xml -> institutionManager.xml
    - replace mongo with jpa dao
    - update spring configuration

cslzchen committed:
Fix logger classs.

cslzchen committed:
Add `scope` and `personalAccessToken` to JPA prototype.
    - update spring config
    - update osf dao
    - @Entity: OpenScienceFrameworkApiOauth2Scope
    - @Entity: OpenScienceFrameworkApiOauth2PersonalAccessToken

cslzchen committed:
Update scope and personal access token hanlder:
    - replace mongo with jpa dao
    - update spring config

cslzchen committed:
Remove all mongo adaptors and update naming. [skip ci]
    - .adaptors.mongo -> .adaptors.postgres.daos/handlers/models
    - subpackage .daos entity mananger
    - subpackage .handlers cas hook
    - subpackage .models: entity class

cslzchen committed:
Update java doc str.

cslzchen committed:
Update TODO comments, fix style. [skip ci]

cslzchen committed:
Remove spring config for mongo.

cslzchen committed:
Add `oauthApplication` to JPA prototype:
    - @Entity: OpenScienceFrameworkApiOauth2Application
    - update OSF DAO
    - update spring config

cslzchen committed:
Update `serviceRegistryDao`:
    - replace mongo with jpa dao
    - update spring config

cslzchen committed:
Remove unused mappings and getters in OSF entites.

cslzchen committed:
Remove TODO for "handling timestamp with time zone for JPA".
    - Django ORM enforce UTC as time zone.
    - JPA ignores time zone.

cslzchen committed:
CAS returns User's PK instead of GUID.

cslzchen committed:
Update model name for `addons_twofactorusersettings` + style fix.

cslzchen committed:
Fix the issue that spring bcrypt does not recognize `$2b`.
    - spring is not vulnerable to OpenBSD's `u_int8_t` vulneralility
    - replace `$2b` with `$2b` before `BCrypt.checkpw()`
    - get exception handler

cslzchen committed:
Add support for Django BCryptSHA256.

cslzchen committed:
Improve code logic for password check.

cslzchen committed:
Update doc str, todo and comments.

cslzchen committed:
Update support for `$2`.

icereval commented on etc/cas.properties:
```diff
@@ -70,6 +70,14 @@ osf.mongodb.dbname=${OSF_DB_NAME:osf20130903}
+# Open Science Framework Postgres Database
+#
+osf.database.driverClass=org.postgresql.Driver
+osf.database.url=jdbc:postgresql://${OSF_DB_PORT_5432_TCP_ADDR:127.0.0.1}:${OSF_DB_PORT_5432_TCP_PORT:5432}/osf?targetServerType=master
```
I'm not sure if we even use the older formatting for docker anymore, if we do not we should go with more standard environment variable names like:

```
osf.database.driverClass=org.postgresql.Driver
osf.database.url=${OSF_DB_URL:jdbc:postgresql://127.0.0.1/osf?targetServerType=master}
osf.database.user=${OSF_DB_USER:postgres}
osf.database.password=${OSF_DB_PASSWORD:}
```

cslzchen committed:
Add custom type StringList for postgres VARCHAR[].

cslzchen committed:
User PostgreSQL82Dialect for OSF.

cslzchen committed:
Use `Types.ARRAY` instead of `Types.VARCHAR` for VARCHAR[]. [skip ci]

cslzchen committed:
Add StringArray UserType, update StringList and OSF User.

cslzchen committed:
Use native postgres query to find user by email.

cslzchen committed:
Refactor StringList/Array to StringList/ArrayUserType.

cslzchen committed:
Refactor OSFInstitution.provider_id to ._id. [skip ci]

cslzchen committed:
Fix style check.

cslzchen committed:
Use varchar instead of uuid for client_id.

cslzchen committed:
Add objectId @ column "_id" for OSFApiOauth2Application.
    - getId() returns objectId instead of postgres id
    - OauthRegisteredService set id by getId() above

cslzchen committed:
Use objectId for "_id" and getId() returns objectId instead of postgres id.

cslzchen committed:
Fix style check.

cslzchen committed:
FindOneUserByEmail includes FindOneUserByUsername.
    - simplify find user in auth handler
    - update docstr, todo and comments

cslzchen committed:
Update reference to `objectId`@"_id" for instituion and oauth application.

cslzchen committed:
Update String[Array|List]UserType.

cslzchen committed:
Enforce password hash prefix and remove logs that may contain plaintext password.
    - update OSFAuthHandler docstr, comments and todo
    - remove unused getter fro OSFUser
    - suppress "unchecked" warning in OSFDAOImpl

cslzchen committed:
Use PostgreSQL82Dialect (OSF) and PostgreSQLDialect(CAS) as default.

cslzchen committed:
Update comments:
    - JPA customized user type
    - Native postgres query for array operations
    - Log that does not leak password

cslzchen commented on etc/cas.properties:
```diff
@@ -70,6 +70,15 @@ osf.mongodb.dbname=${OSF_DB_NAME:osf20130903}
+osf.database.url=jdbc:postgresql://${OSF_DB_PORT_5432_TCP_ADDR:127.0.0.1}:${OSF_DB_PORT_5432_TCP_PORT:5432}/osf?targetServerType=master
+osf.database.user=${CAS_DB_USERNAME:postgres}
+osf.database.password=${CAS_DB_PASSWORD:}
+osf.database.hibernate.dialect=org.hibernate.dialect.PostgreSQL82Dialect
```
[Deploy Note] Add this configuration for OSF Postgres Dialect.

cslzchen committed:
Change `osf_models_...` to `osf_...`.

cslzchen committed:
Update table name: twofactor_twofactorusersettings.

cslzchen committed:
Update table name: twofactor_twofactorusersettings.

Remove column: version.

cslzchen committed:
Update column `_id`.

icereval committed:
rebase

cslzchen committed:
Table name change: addons_twofactor_usersettings.

icereval committed:
Merge branch 'feature/django-osf' of git://github.com/cslzchen/cas-overlay into feature/django-osf

cslzchen committed:
Partially fix conflicts and merge remote-tracking branch
 'upstream/master' into feature/django-osf

cslzchen committed:
Add institution login support:
    - add loginUrl and deleted to institution model
    - add findAllInstitutions to dao
    - update institution handler
    - fully fix merge conflicts

cslzchen committed:
Fix query findOneInstitutionById.

cslzchen committed:
Remove mongo form maven and settings.

cslzchen commented:
<strike>PR closed and new PRs will target dedicated feature [branch](https://github.com/CenterForOpenScience/cas-overlay/tree/feature/django-osf).<strike>

cslzchen committed:
Merge remote-tracking branch 'upstream/feature/django-osf'
 into feature/django-osf.

cslzchen committed:
Merge remote-tracking branch 'upstream/master' into feature/django-osf.

icereval committed:
Merge pull request #32 from cslzchen/feature/django-osf

[Feature] [Django-OSF] Update CAS for Django-OSF [#CAS-5]

cslzchen committed:
Fix wrong query for TFA.

icereval committed:
Merge pull request #34 from cslzchen/feature/django-osf

[Fix] [Django-OSF] Fix Wrong Query for Two-Factor Authentication [No Ticket]

cslzchen committed:
Merge remote-tracking branch 'upstream/master' into feature/django-osf.

cslzchen committed:
Add osf_osfguid and django_content_type to models.

cslzchen committed:
Add query for finding  guid by user to DAOs.

cslzchen committed:
Return guid for service validation and oauth authorization.

cslzchen committed:
Update comment and remove old code.

cslzchen committed:
Fix wrong class name in query.

icereval committed:
Merge pull request #36 from cslzchen/feature/django-osf

[Feature] CAS -->> GUID -->> OSF [#CAS-18]

cslzchen committed:
Fix conficts and port latest develop to django-osf.
    Merge remote-tracking branch 'upstream/develop'
    into feature/django-osf.

cslzchen committed:
Use django "unusable" password instead of null/None.
    - add methods to check if a password hash has a valid
      prefix or is "django-unsuable", which is the only
      two possible settings for a user's password.

cslzchen committed:
Log user status check.

cslzchen committed:
Merge branch 'fix/remove-shib-cookie' of https://github.com/cslzchen/cas-overlay into feature/django-osf.

icereval committed:
Merge branch 'hotfix/0.9.3'

cslzchen committed:
Merge branch 'master' of https://github.com/CenterForOpenScience/cas-overlay into feature/django-osf

icereval committed:
Merge pull request #39 from cslzchen/feature/django-osf

[Feature] Port Recent Changes to Django-OSF [No Ticket]

cslzchen committed:
Merge remote-tracking branch 'upstream/develop' into feature/django-osf.

cslzchen committed:
Merge pull request #40 from cslzchen/feature/django-osf

Sync Commits (No File Change)
